From f7fb9a9597ec1f428455e9ef1cfcdac879a7af83 Mon Sep 17 00:00:00 2001
From: Justinas Grigas <jstn_as@protonmail.com>
Date: Fri, 4 Jun 2021 10:44:24 +0000
Subject: [PATCH] cycles between fonts
---
 x.c      | 14 +++++++++++++-

diff --git a/x.c b/x.c
index 7186040..9a658fa 100644
--- a/x.c
+++ b/x.c
@@ -59,6 +59,7 @@ static void zoom(const Arg *);
 static void zoomabs(const Arg *);
 static void zoomreset(const Arg *);
 static void ttysend(const Arg *);
+static void cyclefonts(const Arg *);

 /* config.h for applying patches and the configuration. */
 #include "config.h"
@@ -328,6 +329,17 @@ ttysend(const Arg *arg)
 	ttywrite(arg->s, strlen(arg->s), 1);
 }

+void
+cyclefonts(const Arg *arg)
+{
+	currentfont++;
+	currentfont %= (sizeof fonts / sizeof fonts[0]);
+	usedfont = fonts[currentfont];
+	Arg larg;
+	larg.f = usedfontsize;
+	zoomabs(&larg);
+}
+
 int
 evcol(XEvent *e)
 {
@@ -1115,7 +1127,7 @@ xinit(int cols, int rows)
 	if (!FcInit())
 		die("could not init fontconfig.\n");

-	usedfont = (opt_font == NULL)? font : opt_font;
+	usedfont = (opt_font == NULL)? fonts[currentfont] : opt_font;
 	xloadfonts(usedfont, 0);

 	/* colors */
--
2.31.1

